### PR TITLE
feature user can delete own post and replies

### PR DESCRIPTION
**Pull request by milack68**

### CODE REVIEW

File: app/Policies/PostPolicy.php [hunk 1]
```diff
@@ -50,7 +51,7 @@ public function update(User $user)
      */
     public function delete(User $user)
     {
-        if ($user->can('delete posts')) {
+        if ($user->can('delete any posts') || ($user->can('delete own posts') && $user == auth()->user())){
```
Comment (RheaMars) [inline app/Policies/PostPolicy.php:54]: The first parameter here is the auth user, so `$user == auth()->user()` will always return true. You need to pass in the post to delete as a second parameter and check if this post belongs to `$user`.

File: app/Policies/PostReplyPolicy.php [hunk 2]
```diff
@@ -50,7 +50,7 @@ public function update(User $user)
      */
     public function delete(User $user)
     {
-        if ($user->can('delete post replies')) {
+        if ($user->can('delete any post replies') || ($user->can('delete own post replies') && $user == auth()->user())) {
```
Comment (RheaMars) [inline app/Policies/PostReplyPolicy.php:53]: As with the PostPolicy: The first parameter here is the auth user.

File: resources/lang/de.json [hunk 3]
```diff
@@ -895,5 +895,7 @@
 	"Delete this reply": "Kommentar löschen",
 	"Post and all replies and comments in this post successfully deleted.": "Der Beitrag und alle Antworten und Kommentare wurden erfolgreich gelöscht",
 	"Post successfully restored.": "Beitrag erfolgreich wiederhergestellt.",
-	"Post successfully updated.": "Beitrag erfolgreich aktualisiert."
```
Comment (RheaMars) [inline resources/lang/de.json:898]: Why did you delete this line, @milack68? Don't we need this anymore?

File: app/Http/Controllers/PostReplyController.php [hunk 4]
```diff
@@ -105,9 +105,10 @@ public function update(PostReplyRequest $request, PostReply $postReply)
     public function destroy(PostReply $postReply)
     {
         $this->authorize('delete', PostReply::class);
+
         $post = $postReply->post_id;
         $postReply->delete();
```
Comment (RheaMars) [inline app/Http/Controllers/PostReplyController.php:110]: I can't see where exactly the fix is for the problem you described, @milack68. ("if a user deletes a reply the whole post and all its replies are deleted"). Did this really happen? Within this method only the postReply gets deleted as it should be, right?

**Review by Cosnavel — CHANGES_REQUESTED** [review]
refs @RheaMars

File: app/Http/Controllers/PostReplyController.php (hunk 4 above)
Comment (milack68) [reply app/Http/Controllers/PostReplyController.php:110]: The problem was in the blade file: by clicking on the delete button for deleting a reply the modal for delete post opend.

File: resources/lang/de.json (hunk 3 above)
Comment (milack68) [reply resources/lang/de.json:898]: You are right. I will add this line once again.
